### ISSUE TITLE
rule 'not-capitalized-test-case-title' throws IndexError, if there is no test case name

### ISSUE THREAD

**Issue by d-biehl** [MarketSquare/robotframework-robocop#333]
Take this file:

```robotframework
*** Test Case ***

#  no test case name!!!
    Log To Console  hello
```

and the rule 'not-capitalized-test-case-title' throws this:
    
     ...
     File "c:\develop\robot\robotframework-robocop\robocop\checkers\naming.py", line 289, in visit_TestCase
        if not node.name[0].isupper():
    IndexError: string index out of range

**Comment by bhirsz** [comment]
Do you think we should add "Missing test case name" rule for this sitatuon? If I run it with robot it returns "Test case name cannot be empty.". although there is chance Robocop return parsing error after applying your fix

**Comment by d-biehl** [comment]
Maybe you should. From robot parsing model, this is a correct file, the "Test case name cannot be empty." The error is only detected at running the file. This is because Robotframework is a simple interpreter.

But there is another situation:

```robotframework
*** Test Case ***
first
    first kw
    
    
*** Keywords ***
# no keyword name here!!!
    Log To Console  hi

first kw
    Log To Console  hello
```

Here we have keyword with out a name, and robot is not reporting an error, because you cannot use this keyword any way. 

I think in both situations, it is a good indication during development that something can throw an error or that something can't be used or is dead code.
